src/System.Buffers/src/System/Buffers/BufferBucket.cs

```diff
@@ -41,7 +41,7 @@ internal T[] Rent()
             if (_index >= _data.Length)
             {
                 // Exit the lock as soon as possible
-                _lock.Exit();
+                _lock.Exit(false);
```

Comment:
Removing the memory barrier should give a perf increase since we remove an interlocked operation whenever exit is called; the side effect is that the fairness of what thread gets the lock next goes away. I think this is an acceptable tradeoff since thread contention on this lock should be low while the performance gain should help out callers more than potential fairness issues.